app/questionnaire/navigation.py

```diff
@@ -40,14 +40,25 @@ def build_navigation(self, current_group_id, current_group_instance):
             logger.debug("Building frontend navigation for group %s", group['id'])
 
             repeating_rule = SchemaHelper.get_repeat_rule(group)
+            skip_group = self._should_skip_group(current_group_instance, group)
+
+            if not skip_group:
```

Comment:
Rather than the nested `if` statement wonder if using `and` and `elif` might read easier?